torch/lib/THD/base/data_channels/DataChannelGloo.cpp

```diff
@@ -149,6 +146,14 @@ void DataChannelGloo::allGatherT(std::vector<thpp::Tensor*>& output,
 
 void DataChannelGloo::allGather(std::vector<thpp::Tensor*>& output,
                                 thpp::Tensor& input, THDGroup group_id) {
+  RETURN_IF_NOT_IN_GROUP(_)
+
+  if (output.size() != _groups.at(group_id).size())
+    throw std::logic_error("allGather: number of output tensors and group size does not match");
+
+  for (auto out_tensor : output)
+    assertTensorEqual(*out_tensor, input, "allGather");
```

Comment:
Here is implementation: [equal](https://github.com/apaszke/pytorch-dist/blob/thd/torch/lib/THD/base/data_channels/DataChannelUtils.hpp#L19)